src/content/lib/utils.js

```diff
@@ -35,30 +36,70 @@ const FAKE_BUGS = [
   },
 ];
 
+function getFakeBugsRequest(bodyString) {
+  return new Request(`/api/bugs?fakebody=${bodyString}`);
+}
+
 export async function runQuery(query) {
   if (prefs.get("offline_debug")) {
     return FAKE_BUGS;
   }
   let data = {};
-  const resp = await fetch("/api/bugs", {
+  let body = JSON.stringify(query);
+  let request = new Request("/api/bugs", {
     headers: {
       Accept: "application/json",
       "Content-Type": "application/json",
     },
-    body: JSON.stringify(query),
+    body,
     method: "POST",
   });
+  let fakeRequest = getFakeBugsRequest(body);
+  let resp = await fetch(request);
+  await cache.set(fakeRequest, resp);
   try {
     data = await resp.json();
   } catch (e) {
     console.log(resp); // eslint-disable-line
     console.log(query); // eslint-disable-line
     console.error(e); // eslint-disable-line
   }
-
   return postProcess(data);
 }
 
+export async function runQueries(queries) {
+  return Promise.all(queries.map(runQuery));
+}
+
+export async function matchQuery(query) {
+  if (prefs.get("offline_debug")) {
+    return FAKE_BUGS;
+  }
+  if (prefs.get("disable_cache")) {
+    return Promise.reject();
+  }
+  let data;
+  let fakeRequest = getFakeBugsRequest(JSON.stringify(query));
+  const response = await cache.get(fakeRequest);
+  if (response) {
+    try {
+      data = await response.json();
+    } catch (e) {
+      console.log("Error parsing cached response :>> ", response); // eslint-disable-line
+      console.log(query); // eslint-disable-line
+      console.error(e); // eslint-disable-line
+    }
+    if (data) {
+      return postProcess(data);
+    }
+  }
+  return data ? postProcess(data) : Promise.reject();
```

Comment:
`data` can't be truthy here because we would have already returned.

`throw new Error(...)`